### PR TITLE
Add JiraTicket button for Many Queries Insight 

### PR DESCRIPTION
**Pull request by opoliarush**
image
image

Bug fixes:
* fixed showing error inside ticket info  for endpoint insights when do link/unlik
* enriched data with commit info from insights

### CODE REVIEW

File: src/components/Insights/tickets/common/useTicketDataSource.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  return {
+    isLoading: isLoading || isInsightMetaIsLoading,
+    codeLocations,
+    spanInsight,
+    commitInfos,
+    onReloadSpanInsight
+  };
+};
+
+export const useSpanDataSource = <TInsight extends GenericCodeObjectInsight>(
```
Comment (kshmidt-digma) [inline src/components/Insights/tickets/common/useTicketDataSource.ts:85]: Consider to split this into 2 separate hooks: `useCodeLocations` and `useCommitInfo`

File: src/components/Insights/HighNumberOfQueriesInsight/index.tsx [hunk 2]
```diff
@@ -25,6 +28,14 @@ export const HighNumberOfQueriesInsight = (
     props.onTraceButtonClick(trace, insightType, spanCodeObjectId);
   };
 
+  const handleCreateJiraTicketButtonClick = () => {
+    sendTrackingEvent(trackingEvents.JIRA_TICKET_INFO_BUTTON_CLICKED, {
+      insightType: insight.type
+    });
+    props.onJiraTicketCreate &&
+      props.onJiraTicketCreate(insight, insight.spanInfo?.spanCodeObjectId);
```
Comment (kshmidt-digma) [inline src/components/Insights/HighNumberOfQueriesInsight/index.tsx:36]: remove 
```suggestion
      props.onJiraTicketCreate(insight);
```

File: src/components/Insights/index.tsx [hunk 3]
```diff
@@ -91,6 +94,20 @@ const renderInsightTicket = (
     );
   }
 
+  if (
+    isEndpointHighNumberOfQueriesInsight(data.insight) &&
+    data.spanCodeObjectId
```
Comment (kshmidt-digma) [inline src/components/Insights/index.tsx:99]:
```suggestion
    isEndpointHighNumberOfQueriesInsight(data.insight)
```

File: src/components/Insights/tickets/common/CommitInfos/types.ts [hunk 4]
```diff
@@ -2,6 +2,6 @@ import { GenericCodeObjectInsight } from "../../../types";
 import { CommitInfosData } from "../../types";
 
 export interface CommitInfosProps {
-  insight: GenericCodeObjectInsight | undefined;
-  commitInfos: CommitInfosData | undefined;
+  insight: GenericCodeObjectInsight | undefined | null;
```
Comment (kshmidt-digma) [inline src/components/Insights/tickets/common/CommitInfos/types.ts:5]:
```suggestion
  insight?: GenericCodeObjectInsight | null;
```

File: src/components/Insights/tickets/common/NPlusOneAffectedEndpoints/types.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { SpanNPlusOneInsight } from "../../../types";
 
 export interface NPlusOneAffectedEndpointsProps {
-  insight: SpanNPlusOneInsight | undefined;
+  insight: SpanNPlusOneInsight | undefined | null;
```
Comment (kshmidt-digma) [inline src/components/Insights/tickets/common/NPlusOneAffectedEndpoints/types.ts:4]:
```suggestion
  insight?: SpanNPlusOneInsight | null;
```

File: src/components/Insights/tickets/common/useTicketDataSource.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
+import { useEffect, useState } from "react";
+import { dispatcher } from "../../../../dispatcher";
+import { InsightType, SpanInfo } from "../../../../types";
+import { actions } from "../../actions";
+import { GenericCodeObjectInsight } from "../../types";
+import { getInsightCommits } from "../getInsightCommits";
+import { CodeLocationsData, CommitInfosData } from "../types";
+
+export const useEndpointDataSource = <
```
Comment (kshmidt-digma) [inline src/components/Insights/tickets/common/useTicketDataSource.ts:9]: Please move all hooks to separate files

**Review by kshmidt-digma — CHANGES_REQUESTED** [review]

File: src/components/Insights/HighNumberOfQueriesInsight/index.tsx (hunk 2 above)
Comment (opoliarush) [reply src/components/Insights/HighNumberOfQueriesInsight/index.tsx]: fixed

File: src/components/Insights/tickets/common/useTicketDataSource.ts (hunk 1 above)
Comment (opoliarush) [reply src/components/Insights/tickets/common/useTicketDataSource.ts]: extracted

File: src/components/Insights/tickets/common/CommitInfos/types.ts (hunk 4 above)
Comment (opoliarush) [reply src/components/Insights/tickets/common/CommitInfos/types.ts]: fxied

File: src/components/Insights/tickets/common/NPlusOneAffectedEndpoints/types.ts (hunk 5 above)
Comment (opoliarush) [reply src/components/Insights/tickets/common/NPlusOneAffectedEndpoints/types.ts]: fixed

File: src/components/Insights/tickets/common/useTicketDataSource.ts (hunk 6 above)
Comment (opoliarush) [reply src/components/Insights/tickets/common/useTicketDataSource.ts]: done

**Review by kshmidt-digma — APPROVED** [review]

**Pull request merged**